### ISSUE TITLE
Loader run twice when using extract-text-webpack-plugin

### ISSUE THREAD

**Issue by milewski** [webpack-contrib/extract-text-webpack-plugin#647]
Im developing a postcss plugin and i found out that the loader runs twice when used in conjunction with `extract-text-webpack-plugin`

```
"extract-text-webpack-plugin": "3.0.1",
"webpack": "3.6.0"
```

and adding `allChunks: true` to the options seems to make it run only once.... 

Reproducible test repository: https://github.com/milewski/extract-text-webpack-plugin-bug 

@d3viant0ne

**Comment by stereokai** [comment]
@d3viant0ne I am new to Webpack. In our build, I'm using Webpack 3.6 and the latest version of all loaders in use. I am not doing anything out of the ordinary - only standard stuff, loading JS, CSS, images and fonts, and still getting duplicate outputs in my console when I am using ExtractTextPlugin. Without ExtractTextPlugin, there is no duplicate output. If I can do anything to help find out where the issue lies, please let me know.

**Comment by d3viant0ne** [comment]
@stereokai - @milewski has already done most of the work with the minimal reproduction that clearly highlights the issue.

There isn't much else to do at this point besides `node  --inspect --inspect-brk --stack-size=8024 node_modules/webpack/bin/webpack.js` and going hunting.

Given `allChunks: true` resolves the issue & what that setting actually does, I am pretty sure what's going on but you never really know until you get into it.

**Comment by milewski** [comment]
Entry
```js
entry: {
    'app': './app.js',
    'global': './global.scss',
    'component': './component.scss'
},
```

Custom Plugin
```js
apply(compiler: Compiler): void {

    compiler.plugin('compilation', compilation => {
        console.log(compilation.name)
    })

}
```

Outputs:

```
undefined //which i guess is the main webpack compilation... 

extract-text-webpack-plugin G:\Code\node_modules\extract-text-webpack-plugin\dist G:\Code\node_modules\css-loader\index.js??ref--1-2!G:\Code\icon-fon
t-plugin\node_modules\postcss-loader\lib\index.js??ref--1-3!G:\Code\node_modules\resolve-url-loader\index.js!G:\Code\node_modules\sass-loader\lib\loa
der.js??ref--1-5!G:\Code\demo\global.scss

extract-text-webpack-plugin G:\Code\node_modules\extract-text-webpack-plugin\dist G:\Code\node_modules\css-loader\index.js??ref--1-2!G:\Code\icon-fon
t-plugin\node_modules\postcss-loader\lib\index.js??ref--1-3!G:\Code\node_modules\resolve-url-loader\index.js!G:\Code\node_modules\sass-loader\lib\loa
der.js??ref--1-5!G:\Code\demo\component.scss
```

Seems that extract-text-webpack-plugin makes webpack triggers `compilation` for each .scss file you have as entry point so in this case i had 2, then it triggered 3 times, causing every single plugin that listen to any compilation event triggers `X * Number of .scss files extracted` :O

**Comment by stereokai** [comment]
@milewski @d3viant0ne Exactly what I observed

**Comment by alexander-akait** [comment]
DEPRECATED
Please use: https://github.com/webpack-contrib/mini-css-extract-plugin
